h2/src/main/org/h2/table/Plan.java

```diff
@@ -38,8 +37,8 @@
     public Plan(TableFilter[] filters, int count, Expression condition) {
         this.filters = new TableFilter[count];
         System.arraycopy(filters, 0, this.filters, 0, count);
-        final ArrayList<Expression> allCond = Utils.newSmallArrayList();
-        final ArrayList<TableFilter> all = Utils.newSmallArrayList();
+        final ArrayList<Expression> allCond = new ArrayList<>(count/2);
+        final ArrayList<TableFilter> all = new ArrayList<>(count);
```

Comment:
These size assumptions are also incorrect. They are too small in about 43% of cases from the `TestAll`. It looks like plain `new ArrayList<>()` with its default initial capacity of 10 is the most reasonable choice here.